common/djangoapps/course_modes/admin.py

```diff
@@ -43,38 +96,111 @@ def clean_course_id(self):
-            expiration_datetime = self.instance.expiration_datetime.replace(tzinfo=None)
-            self.initial['expiration_datetime'] = default_tz.localize(expiration_datetime)
-
     def clean_expiration_datetime(self):
-        """changing the tzinfo for a given datetime object"""
+        """
+        Ensure that the expiration datetime we save uses the UTC timezone.
+        """
         # django admin saving the date with default timezone to avoid time conversion from form to db
         # changes its tzinfo to UTC
         if self.cleaned_data.get("expiration_datetime"):
             return self.cleaned_data.get("expiration_datetime").replace(tzinfo=UTC)
 
+    def clean_verification_deadline(self):
+        """
+        Ensure that the verification deadline we save uses the UTC timezone.
+        """
+        if self.cleaned_data.get("verification_deadline"):
+            return self.cleaned_data.get("verification_deadline").replace(tzinfo=UTC)
+
+    def clean(self):
+        """
+        Clean the form fields.
+        This is the place to perform checks that involve multiple form fields.
+        """
+        cleaned_data = super(CourseModeForm, self).clean()
+        mode_slug = cleaned_data["mode_slug"]
+        upgrade_deadline = cleaned_data["expiration_datetime"]
+        verification_deadline = cleaned_data["verification_deadline"]
+
+        # Allow upgrade deadlines ONLY for the "verified" mode
+        # This avoids a nasty error condition in which the upgrade deadline is set
+        # for a professional education course before the enrollment end date.
+        # When this happens, the course mode expires and students are able to enroll
+        # in the course for free.  To avoid this, we explicitly prevent admins from
+        # setting an upgrade deadline for any mode except "verified" (which has an upgrade path).
+        if upgrade_deadline is not None and mode_slug != CourseMode.VERIFIED:
+            raise forms.ValidationError(
+                'Only the "verified" mode can have an upgrade deadline.  '
+                'For other modes, please set the enrollment end date in Studio.'
+            )
+
+        # Verification deadlines are allowed only for verified modes
+        if verification_deadline is not None and mode_slug not in CourseMode.VERIFIED_MODES:
+            raise forms.ValidationError("Verification deadline can be set only for verified modes.")
+
+        # Verification deadline must be after the upgrade deadline,
+        # if an upgrade deadline is set.
+        # There are cases in which we might want to set a verification deadline,
+        # but not an upgrade deadline (for example, a professional education course that requires verification).
+        if verification_deadline is not None:
+            if upgrade_deadline is not None and verification_deadline < upgrade_deadline:
+                raise forms.ValidationError("Verification deadline must be after the upgrade deadline.")
```

Comment:
Note to reviewers: these are the additional validation checks.  Eventually these should be enforced by the Course Admin Tool.